corehq/apps/auditcare/tasks.py

```diff
@@ -0,0 +1,102 @@
+import logging
+
+from couchdbkit.ext.django.loading import get_db
+
+from dimagi.utils.dates import force_to_datetime
+from dimagi.utils.logging import notify_exception
+
+from corehq.apps.auditcare.models import (
+    ACCESS_LOGIN,
+    ACCESS_LOGOUT,
+    ACCESS_FAILED,
+    AccessAudit,
+    NavigationEventAudit,
+)
+from corehq.util.soft_assert import soft_assert
+from corehq.apps.auditcare.utils.migration import AuditCareMigrationUtil, get_formatted_datetime_string
+
+log = logging.getLogger(__name__)
+
+ACCESS_LOOKUP = {
+    "login": ACCESS_LOGIN,
+    "logout": ACCESS_LOGOUT,
+    "login_failed": ACCESS_FAILED,
+}
+
+
+def get_couch_key(time):
+    return [time.year, time.month, time.day, time.hour, time.minute, time.second]
+
+
+def get_migration_key(start_time, end_time):
+    return get_formatted_datetime_string(start_time) + '_' + get_formatted_datetime_string(end_time)
+
+
+def copy_events_to_sql(start_time, end_time, retry=0):
+    db = get_db("auditcare")
+    util = AuditCareMigrationUtil()
+    log.info(f"Starting batch: {start_time} - {end_time}")
+    print(f"Starting batch: {start_time} - {end_time}")
+    key = get_migration_key(start_time, end_time)
+    endkey = get_couch_key(end_time)
+    startkey = get_couch_key(start_time)
+    util.log_batch_start(key)
+    count = 0
+    is_errored = False
+    for result in db.view(
+        "auditcare/all_events",
+        startkey=startkey,
+        endkey=endkey,
+        reduce=False,
+        include_docs=True,
+    ):
```

Comment:
This view call will fetch all rows from the beginning to the end of the batch in a single result set. It would be more efficient and would support large ranges if it instead queried batches of N records (using `limit=N` where N is something like 1000 or 10000). The start key for the next batch should be the last key returned in the previous batch. Continue querying batches until you get a result with < N records, which indicates the last batch.

Also, retry logic should also be used on Couch queries since Couch frequently fails. There is a decorator for that:
https://github.com/dimagi/commcare-hq/blob/04014d64c711f0a5413dd58510d8a3437317d803/corehq/ex-submodules/dimagi/utils/couch/database.py#L191-L199